nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-0-10-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/ConsumerPool.java

```diff
@@ -126,6 +128,25 @@ public ConsumerLease obtainConsumer(final ProcessSession session) {
         return lease;
     }
 
+    public void retainConsumers() {
+        pooledLeases.forEach(lease -> {
```

Comment:
Is there any issue if another thread calls obtainConsumer while retainConsumers is in the middle of looping over the consumers?